### PR TITLE
Feature/main page refactoring

### PR DESCRIPTION
**Pull request by o-mehlii**

### CODE REVIEW

File: renderer/src/helpers/hooks/useTheme.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+const useTheme = () => {
+  const [isOSDarkTheme, setIsOSDarkTheme] = useState(true);
+
+  const [theme, setTheme] = useThemeStore(
+    (state) => [state.theme, state.setTheme],
+    shallow
+  );
+
+  function handleThemeChange(e) {
```
Comment (okhmelevskyi) [inline renderer/src/helpers/hooks/useTheme.ts:13]: Could you, please, overwrite this function?
Example` const handleThemeChange = (e) = >  e.matches ?  setIsOSDarkTheme(true) :  setIsOSDarkTheme(false);`
OR
You could test e.matches inside `setIsOSDarkTheme(Boolean(e.matches))` - it should be tested.

File: renderer/src/components/MainPage/MainPage.tsx [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+      setSelectedDateActivities(parsedActivities);
+      return;
+    }
+
+    setReportAndNotes([]);
+    setSelectedDateActivities([]);
+  }, [selectedDateReport]);
+
+  useEffect(() => {
+    (async () => {
```
Comment (okhmelevskyi) [inline renderer/src/components/MainPage/MainPage.tsx:90]: May I ask you to move the logic from the useEffect hook to a new function and invoke the function inside that useEffect hook? Name the function with a sense, please.

**Review by okhmelevskyi — CHANGES_REQUESTED** [review]

File: renderer/src/components/MainPage/MainPage.tsx [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+      const dayReport = await global.ipcRenderer.invoke(IPC_MAIN_CHANNELS.READ_DAY_REPORT, reportsFolder, selectedDate);
+
+      setSelectedDateReport(dayReport || "");
+    })();
+
+    global.ipcRenderer.send(IPC_MAIN_CHANNELS.START_FILE_WATCHER, reportsFolder, selectedDate);
+    global.ipcRenderer.on("file-changed", (event, data) => {
```
Comment (a-nahaitsev) [inline renderer/src/components/MainPage/MainPage.tsx:97]: move `file-changed` to constants

File: renderer/src/components/MainPage/MainPage.tsx [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+      if (selectedDateReport != data) {
+        setSelectedDateReport(data || "");
+      }
+    });
+    global.ipcRenderer.on("errorMes", (event, data) => {
```
Comment (a-nahaitsev) [inline renderer/src/components/MainPage/MainPage.tsx:102]: i guess, the same

File: renderer/src/components/MainPage/MainPage.tsx [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+      console.log("event ", event);
+      console.log("data ", data);
+    });
+
+    return () => {
+      global.ipcRenderer.removeAllListeners("file-changed");
+      global.ipcRenderer.removeAllListeners("errorMes");
+      global.ipcRenderer.send(IPC_MAIN_CHANNELS.STOP_PATH_WATCHER, reportsFolder, selectedDate);
```
Comment (a-nahaitsev) [inline renderer/src/components/MainPage/MainPage.tsx:110]: i guess, the same here

**Review by a-nahaitsev — CHANGES_REQUESTED** [review]

File: renderer/src/components/MainPage/MainPage.tsx [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+  setSelectedDate,
+  latestProjAndAct,
+  setTrackTimeModalActivity,
+}: MainPageProps) => {
+  const [isDropboxConnected, setIsDropboxConnected] = useState(true);
+  const [calendarDate, setCalendarDate] = useState(new Date());
+  const [reportAndNotes, setReportAndNotes] = useState<any[] | ReportAndNotes>([]);
+  const [selectedDateReport, setSelectedDateReport] = useState("");
+  const { screenSizes } = useScreenSizes();
+  const isManualInputMain = localStorage.getItem("is-manual-input-main-section") === "true";
```
Comment (a-nahaitsev) [inline renderer/src/components/MainPage/MainPage.tsx:35]: Move to constants `is-manual-input-main-section`

File: renderer/src/components/MainPage/MainPage.tsx [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+  };
+
+  const handleSave = (report: string, shouldAutosave: boolean) => {
+    setSelectedDateReport(report);
+    setShouldAutosave(shouldAutosave);
+  };
+
+  const saveSerializedReport = (serializedReport: string) => {
+    global.ipcRenderer.send(IPC_MAIN_CHANNELS.CHECK_DROPBOX_CONNECTION);
+    global.ipcRenderer.invoke("app:write-day-report", reportsFolder, selectedDate, serializedReport);
```
Comment (a-nahaitsev) [inline renderer/src/components/MainPage/MainPage.tsx:119]: move to constants `app:write-day-report`

File: renderer/src/components/SelectFolderPlaceholder/SelectFolderPlaceholder.tsx [hunk 8]
```diff
@@ -1,12 +1,17 @@
...
+const SelectFolderPlaceholder = () => {
+  const [reportsFolder, setReportsFolder] = useMainStore(
+    (state) => [state.reportsFolder, state.setReportsFolder],
+    shallow,
+  );
   const handleButtonClick = () => {
     global.ipcRenderer.invoke("app:select-folder").then((folder: string | null) => {
       if (folder) {
-        setFolder(folder);
+        setReportsFolder(folder);
```
Comment (a-nahaitsev) [inline renderer/src/components/SelectFolderPlaceholder/SelectFolderPlaceholder.tsx:14]: `app:select-folder` on the line 12 - move it to constants as well

**Review by a-nahaitsev — CHANGES_REQUESTED** [review]

**Review by okhmelevskyi — APPROVED** [review]

**Review by a-nahaitsev — APPROVED** [review]

**Pull request merged**